### PR TITLE
TW-1834: profile image is not updated in multiple account

### PR DESCRIPTION
**Pull request by sherlockvn**
### Issue:
#1834 

### Root cause:
- The avatar in the chat list screen have not listen for uploading new avatar, so only when you terminate the app, the avatar will be updated correctly
### Solution:
- We can listen for uploading new avatar using onAccountData stream from matrix client. We also need to put it in the `didUpdateWidget` method, because when switch account, we need to listen to the right client
- The other change is for improve the UX a bit when uploading new avatar

### Demo android:

https://github.com/linagora/twake-on-matrix/assets/43041967/41452fd2-a0f9-4e86-ac9f-3789d3386aa5

### demo ios:

https://github.com/linagora/twake-on-matrix/assets/43041967/30a1f25e-f751-461d-bd54-05d0f38319b6

### CODE REVIEW

**Comment by Te-Z** [comment]
can you add the link of the related issue in the description please ?

**Comment by hieutbui** [comment]
can you also explain a bit about the root cause and solution?

**Comment by sherlockvn** [comment]
> can you also explain a bit about the root cause and solution?

Donee

**Comment by Te-Z** [comment]
Can you demo the display name change also please ? 
https://github.com/linagora/twake-on-matrix/issues/1834#issuecomment-2180166314

**Comment by sherlockvn** [comment]
> Can you demo the display name change also please ? [#1834 (comment)](https://github.com/linagora/twake-on-matrix/issues/1834#issuecomment-2180166314)

https://github.com/linagora/twake-on-matrix/assets/43041967/b8ee6bc2-8a64-432d-a89c-c457209ffe78

File: lib/pages/settings_dashboard/settings_profile/settings_profile.dart [hunk 1]
```diff
@@ -472,12 +472,10 @@ class SettingsProfileController extends State<SettingsProfile>
           'SettingsProfile::_handleUploadProfileOnData() - success: $success',
         );
         if (success is UpdateProfileSuccess) {
-          _clearImageInLocal();
           final newProfile = Profile(
             userId: client.userID!,
             displayName: success.displayName ?? displayName,
-            avatarUrl:
-                success.avatar == null ? currentProfile?.avatarUrl : null,
```
Comment (nqhhdev) [inline lib/pages/settings_dashboard/settings_profile/settings_profile.dart:480]: Pls add case upload avt or display name failed

**Review by hieutbui — APPROVED** [review]

File: lib/pages/settings_dashboard/settings_profile/settings_profile.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/settings_dashboard/settings_profile/settings_profile.dart:480]: done

**Comment by sherlockvn** [comment]
Fix up the missing case:
- Update avatar and display name in multiple account selection picker

File: lib/pages/settings_dashboard/settings_profile/settings_profile.dart [hunk 2]
```diff
@@ -68,6 +72,8 @@ class SettingsProfileController extends State<SettingsProfile>
   AssetEntity? assetEntity;
   FilePickerResult? filePickerResult;
 
+  List<TwakeChatPresentationAccount> multipleAccounts = [];
```
Comment (hieutbui) [inline lib/pages/settings_dashboard/settings_profile/settings_profile.dart:75]: should it private?

Comment (sherlockvn) [reply lib/pages/settings_dashboard/settings_profile/settings_profile.dart]: done

**Pull request merged**